inttest/common/util.go

```diff
@@ -56,6 +57,45 @@ func WaitForKubeRouterReady(ctx context.Context, kc *kubernetes.Clientset) error
 	return WaitForDaemonSet(ctx, kc, "kube-router")
 }
 
+// WaitForCoreDNSReady waits to see all coredns pods healthy as long as the context isn't canceled.
+// It also waits to see all the related svc endpoints to be ready to make sure coreDNS is actually
+// ready to serve requests.
+func WaitForCoreDNSReady(ctx context.Context, kc *kubernetes.Clientset) error {
+	err := WaitForDeployment(ctx, kc, "coredns", "kube-system")
+	if err != nil {
+		return err
+	}
+	// Wait till we see the svc endpoints ready
+	return wait.PollImmediateUntilWithContext(ctx, 100*time.Millisecond, func(ctx context.Context) (bool, error) {
+		epSlices, err := kc.DiscoveryV1().EndpointSlices("kube-system").List(ctx, metav1.ListOptions{
+			LabelSelector: "k8s-app=kube-dns",
+		})
+
+		// NotFound is ok, it might not be created yet
+		if err != nil && !apierrors.IsNotFound(err) {
+			return true, err
```

Comment:
does it mean, that is err is NotFound we don't retry anymore times? 

https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediateUntilWithContext

https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ConditionWithContextFunc

Reply:
> does it mean, that is err is NotFound we don't retry anymore times?

Nope. It means that if the err is NOT `nil` and NOT `NotFound`, then we won't retry and return the err.